api/oidc.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+)
+
+var ErrAudienceTooLong = errors.New("the API only supports at most one element in the audience")
+
+type OidcToken struct {
+	Token string `json:"token"`
+}
+
+func (c *Client) OidcToken(jobId string, audience ...string) (*OidcToken, *Response, error) {
+	type oidcTokenRequest struct {
+		Audience string `json:"audience,omitempty"`
+	}
+
+	m := &oidcTokenRequest{}
+	switch len(audience) {
+	case 0:
+	case 1:
+		m.Audience = audience[0]
```

Comment:
I initially read this as C-style switch fall-through, where cases `0` and `1` are handled the same, and that left me wondering why it didn't panic on `audience[0]` for an empty slice.

Maybe it's worth making it clearer that `case 0:` is a no-op?

(Alternatively, maybe the complexity of handling multiple audiences is YAGNI since we may never support that?)

Reply:
I didn't consider it to be YAGNI as it's in the spec, and we don't control whether 3rd parties would read the spec and think it would be a good idea to require multiple audiences on a OIDC token they consume. Then our implementation would not work with such a service.

But this it not really the right place to have a debate about it, bk/bk needs to support it too. So I'll implement it with at most audience in the agent for now.